source/interface/azure_iot_jws.h

```diff
@@ -15,6 +15,19 @@
 
 #include "azure_iot_result.h"
 
+/**
+ * @brief Holds the values of the root key used to verify the JWS signature.
+ */
+typedef struct AzureIoTJWS_RootKey
+{
+    uint8_t * pucRootKeyId;
```

Comment:
What is the root key ID used for?

Reply:
https://github.com/Azure/azure-iot-middleware-freertos/blob/bf6af17692525dc58e5cc01f8ac1b3d5c11aa00f/ports/mbedTLS/azure_iot_jws_mbedtls.c#L624

The root key has an id within the JWS. We check to make sure the JWS was signed with the root key which we pass to the `AzureIoTJWS_ManifestAuthenticate()` api

Reply:
Shouldn't that check use `pucRootKeyId` from this struct instead?

Reply:
Yes sorry this was pasted from the feature branch. In this PR it's changed.